romanisim/nonlinearity.py

```diff
@@ -199,35 +148,19 @@ class NL:
     tightly coupled to the apportionment mechanism and so I haven't
     explored that adequately.
     """
-    def __init__(self, coeffs):
+    def __init__(self, coeffs, gain=1.0, inverse=False):
         """Construct an NL class handling non-linearity correction.
 
         Parameters
         ----------
         coeffs : np.ndarray[ncoeff, nx, ny] (float)
             Non-linearity coefficients from reference files.
         """
+        self.inverse = inverse
         self.coeffs = repair_coefficients(coeffs)
-        self.derivcoeffs = derivative(self.coeffs)
-        self.coeffs = self.coeffs[::-1, ...].copy()
-        self.derivcoeffs = self.derivcoeffs[::-1, ...].copy()
-
-    def efficiency(self, counts):
-        """Compute the efficiency of photon counting given a count level.
+        self.gain = gain
 
-        Parameters
-        ----------
-        counts : np.ndarray[nx, ny] (float)
-            The counts in each pixel
-
-        Returns
-        -------
-        efficiency : np.ndarray[nx, ny] (float)
-            The efficiency of each pixel at the current count level.
-        """
-        return efficiency(counts, self.derivcoeffs, reversed=True)
-
-    def correct(self, counts):
+    def apply(self, counts, electrons=False, reversed=False):
```

Comment:
electrons needs documentation.  It's what you should use if 'counts' is actually in electrons but the NL coefficients are designed for DN, applying and unapplying the gain.

Reply:
Added.